freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/salt/rotatepassword/RotateSaltPasswordAction.java

```diff
@@ -0,0 +1,43 @@
+package com.sequenceiq.freeipa.flow.freeipa.salt.rotatepassword;
+
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.springframework.statemachine.StateContext;
+
+import com.sequenceiq.cloudbreak.common.event.Payload;
+import com.sequenceiq.flow.core.FlowParameters;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.flow.freeipa.salt.rotatepassword.event.RotateSaltPasswordFailureResponse;
+import com.sequenceiq.freeipa.flow.freeipa.salt.rotatepassword.event.RotateSaltPasswordReason;
+import com.sequenceiq.freeipa.flow.stack.AbstractStackAction;
+import com.sequenceiq.freeipa.service.stack.StackService;
+
+public abstract class RotateSaltPasswordAction<P extends Payload>
+        extends AbstractStackAction<RotateSaltPasswordState, RotateSaltPasswordEvent, RotateSaltPasswordContext, P> {
+
+    public static final String REASON = "reason";
+
+    @Inject
+    private StackService stackService;
+
+    protected RotateSaltPasswordAction(Class<P> payloadClass) {
+        super(payloadClass);
+    }
+
+    @Override
+    protected RotateSaltPasswordContext createFlowContext(FlowParameters flowParameters,
+            StateContext<RotateSaltPasswordState, RotateSaltPasswordEvent> stateContext, P payload) {
+        Stack stack = stackService.getByIdWithListsInTransaction(payload.getResourceId());
```

Comment:
@topolyai5 please check if this now the expected stack fetching

Reply:
freeipa stacks do not have StackView, so it can not be optimized right now